docs/source/topics/data_types.rst

```diff
@@ -492,8 +495,92 @@ Example of creating an ``PortableCode``:
+
+* ``image_name``: The name of the container image (e.g., ``docker://alpine:3``).
+* ``filepath_executable``: The filepath of the executable within the container (e.g. ``/usr/bin/bash``).
+* ``engine_command``: The bash command to invoke the container image (e.g. ``singularity exec --bind $PWD:$PWD {image_name}``).
+  The exact form of this command will depend on the containerization technology that is used.
+* ``computer``: The :class:`~aiida.orm.computers.Computer` on which to run the container.
+
+.. important::
+
+    If the ``engine_command`` contains variables (such as in the ``singularity exec --bind $PWD:$PWD {image_name}`` example), it is crucial that the ``Computer`` needs to have the ``use_double_quotes`` setting set to ``True``.
+    By default, a ``Computer`` will use single quotes to escape command line arguments and so the ``$PWD`` would not be expanded.
+    If this wasn't defined as such when the ``Computer`` was created, it can be changed through the API:
+
+    .. code-block::
+
+        computer = load_computer('some-computer')
+        computer.set_use_double_quotes(True)
+
+.. note::
+
+    The container image can be already download and stored on the target computer or can reference an image on a container index, such as Docker Hub.
+    In the latter case, the first time that the containerized code is run, the image will be automatically pulled from the index, which may take a while if the image is large.
+
+
+.. _topics:data_types:core:code:installed:containerized:setup:
+
+Setup
+^^^^^
+
+A ``ContainerizedCode`` can be created through the CLI as well as the API.
+The following examples show how to setup running ``bash`` in a base Docker container through Singularity to be run on the ``Computer`` named ``some-computer``:
+
+.. tab-set::
+
+    .. tab-item:: CLI
+
+        .. code-block:: console
+
+            verdi code create core.code.containerized \
+                --non-interactive \
+                --label containerized-code \
+                --default-calc-job-plugin core.arithmetic.add \
+                --computer some-computer \
+                --filepath-executable "/bin/sh" \
+                --image-name "docker://alpine:3" \
+                --engine-command "singularity exec --bind $PWD:$PWD {image_name}"
+
+    .. tab-item:: API
+
+        .. code-block:: python
+
+            from aiida.orm import ContainerizedCode, load_computer
+
+            code = ContainerizedCode(
+                computer=load_computer('some-computer')
+                filepath_executable='/bin/sh'
+                image_name='docker://alpine:3',
+                engine_command='singularity exec --bind $PWD:$PWD {image_name}'
+            ).store()
+
+
+.. _topics:data_types:core:code:installed:containerized:support:
+
+Supported container technologies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``ContainerizedCode`` is compatible with a variety of containerization technologies:
```

Comment:
In order to make this subsection most useful, please add for each engine an example of what to put into the `engine_command` field.

Reply:
I agree, and I already put this in [my comment](https://github.com/aiidateam/aiida-core/pull/5667#issuecomment-1282274083). I didn't add these examples since they weren't there in the original PR and I have no idea what they should be. I guess if @unkcpz already has them, we can quickly add them. But if that requires more work, maybe this can be done at a later stage.

Reply:
I add the command to line below.